### PR TITLE
Bael 4805 Decode a JWT Token in Java

### PR DESCRIPTION
**Pull request by LiamGve**

### CODE REVIEW

File: jjwt/src/test/java/io/jsonwebtoken/jjwtfun/util/JWTDecoderUtilUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+    @Test
+    @DisplayName("Given a simple token with no signature, when decoding, then string of header and payload are returned.")
+    void givenSimpleToken_whenDecoding_thenStringOfHeaderPayloadAreReturned() {
+        assertThat(JWTDecoderUtil.decodeJWTToken(SIMPLE_TOKEN))
+          .contains(SignatureAlgorithm.HS256.getValue());
+    }
+
+    @Test
+    @DisplayName("Given a signed token, when decoding with an invalid secret, then the integrity is not validated.")
```
Comment (ashleyfrieze) [inline jjwt/src/test/java/io/jsonwebtoken/jjwtfun/util/JWTDecoderUtilUnitTest.java:23]: Nice use of `@DisplayName` but it's a little unnecessary, given the name of the test also says this. Let's drop it.

**Review by ashleyfrieze — COMMENTED** [review]
Great code!

File: jjwt/src/test/java/io/jsonwebtoken/jjwtfun/util/JWTDecoderUtilUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+class JWTDecoderUtilUnitTest {
+
+    private final static String SIMPLE_TOKEN = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkJhZWxkdW5nIFVzZXIiLCJpYXQiOjE1MTYyMzkwMjJ9";
+    private final static String SIGNED_TOKEN = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkJhZWxkdW5nIFVzZXIiLCJpYXQiOjE1MTYyMzkwMjJ9.qH7Zj_m3kY69kxhaQXTa-ivIpytKXXjZc1ZSmapZnGE";
+
+    @Test
+    void givenSimpleToken_whenDecoding_thenStringOfHeaderPayloadAreReturned() {
+        assertThat(JWTDecoderUtil.decodeJWTToken(SIMPLE_TOKEN))
+          .contains(SignatureAlgorithm.HS256.getValue());
```
Comment (ashleyfrieze) [inline jjwt/src/test/java/io/jsonwebtoken/jjwtfun/util/JWTDecoderUtilUnitTest.java:18]: I imagine this is searching for the substring`HS256` - that's not a very transparent assertion. Can we not assert for something more meaningful to the reader - e.g. a user id claim?

Comment (LiamGve) [reply jjwt/src/test/java/io/jsonwebtoken/jjwtfun/util/JWTDecoderUtilUnitTest.java:18]: Good point, I have updated to the 'name' property in the payload which is "Baeldung User".

**Pull request merged**